module/zfs/ddt_stats.c

```diff
@@ -36,24 +36,28 @@ static void
 ddt_stat_generate(ddt_t *ddt, ddt_entry_t *dde, ddt_stat_t *dds)
 {
 	spa_t *spa = ddt->ddt_spa;
-	ddt_phys_t *ddp = dde->dde_phys;
 	ddt_key_t *ddk = &dde->dde_key;
 	uint64_t lsize = DDK_GET_LSIZE(ddk);
 	uint64_t psize = DDK_GET_PSIZE(ddk);
 
 	memset(dds, 0, sizeof (*dds));
 
-	for (int p = 0; p < DDT_PHYS_TYPES; p++, ddp++) {
+	for (int p = 0; p < DDT_NPHYS(ddt); p++) {
+		const ddt_univ_phys_t *ddp = dde->dde_phys;
+		ddt_phys_variant_t v = DDT_PHYS_VARIANT(ddt, p);
+
 		uint64_t dsize = 0;
-		uint64_t refcnt = ddp->ddp_refcnt;
+		uint64_t refcnt = ddt_phys_refcnt(ddp, v);
```

Comment:
Since ddt_phys_refcnt() is now a function and can't be optimized by compiler, I would move its call down after the branching, just when it is needed.

Reply:
Yeah, nice. Done!